### PR TITLE
fix: don't return gasUsed when estimateGas went wrong

### PR DESCRIPTION
**Pull request by RetricSu**
when a transaction in eth_estimateGas failed in godwoken, web3 will still return the gasUsed to user, while ethereum just simply throw error message. this pr fix web3 to stay same behavior with ethereum.

### CODE REVIEW

File: packages/api-server/src/methods/gw-error.ts [hunk 1]
```diff
@@ -234,3 +234,29 @@ export function failedReasonByErrorReceipt(
...
+  if (gwErr.statusReason != null) {
+    failedReason.message = gwErr.statusReason;
+  }
+  let errorData: any = undefined;
+  if (Object.keys(failedReason).length !== 0) {
+    errorData = { failed_reason: failedReason };
+  }
+
+  let errorMessage = gwErr.message;
+  if (gwErr.statusReason != null && failedReason.status_type != null) {
```
Comment (keroro520) [inline packages/api-server/src/methods/gw-error.ts:254]: When `evmcCodeTypeMapping` doesn't contains the `gwErr.statusCode`, the `failedReason.status_type` will be assigned to `undefined` but not `null`.

Comment (keroro520) [reply packages/api-server/src/methods/gw-error.ts:254]: Should we use `if (!gwErr.statusReason && !failedReason.status_type)` instead?

**Review by keroro520 — APPROVED** [review]

File: packages/api-server/src/methods/gw-error.ts (hunk 1 above)
Comment (RetricSu) [reply packages/api-server/src/methods/gw-error.ts:254]: != null will check undefined and null

**Review by Flouse — APPROVED** [review]

File: packages/api-server/src/methods/gw-error.ts (hunk 1 above)
Comment (keroro520) [reply packages/api-server/src/methods/gw-error.ts:254]: Yeah, you are right. Magic juggling-check 😂

**Review by classicalliu — APPROVED** [review]

**Pull request merged**